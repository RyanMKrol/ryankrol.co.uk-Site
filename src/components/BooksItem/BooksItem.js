import React, { Component } from 'react';

import styles from './BooksItem.module.css';

/**
 * A component representing a books item
 */
class BooksItem extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    this.title = props.title;
    this.authors = props.authors.join(', ');
    this.images = props.images;
    this.isbn = props.isbn;
    this.link = `https://www.amazon.co.uk/s?k=${this.isbn}`;
    this.series = props.series;
    this.numberInSeries = props.numberInSeries;

    if (
      typeof this.title === 'undefined'
      || typeof this.authors === 'undefined'
      || typeof this.images === 'undefined'
      || typeof this.isbn === 'undefined'
      || typeof this.link === 'undefined'
      || typeof this.series === 'undefined'
      || typeof this.numberInSeries === 'undefined'
    ) {
      throw new Error('Could not create BooksItem');
    }
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX BooksItem component
   */
  render() {
    return (
      <div className={styles.container}>
        <img
          className={styles.thumbnail}
          src={this.images.thumbnail}
          alt={`${this.title} - ${this.authors}`}
        />
      </div>
    );
  }
}

export default BooksItem;
