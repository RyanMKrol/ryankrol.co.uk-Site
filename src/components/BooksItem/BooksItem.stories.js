import React from 'react';

import DEFAULT_MOCK from './__mocks__/BooksItemMocks';

export default {
  title: 'Components/BooksItem',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the BooksItem
 */
export const normal = () => DEFAULT_MOCK;
