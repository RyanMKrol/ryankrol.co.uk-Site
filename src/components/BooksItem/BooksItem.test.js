import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import BooksItem from './BooksItem';

import DEFAULT_MOCK from './__mocks__/BooksItemMocks';

it('renders BooksItem without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('renders BooksItem without throwing', () => {
  expect(() => mount(DEFAULT_MOCK)).not.toThrow();
});

it('BooksItem throws without title', () => {
  expect(() => mount(<BooksItem />)).toThrow();
});

it('BooksItem throws without authors', () => {
  expect(() => mount(<BooksItem title="The Humans" />)).toThrow();
});

it('BooksItem throws without images', () => {
  expect(() => mount(<BooksItem authors={['Matt Haig']} title="The Humans" />)).toThrow();
});

it('BooksItem throws without isbn', () => {
  expect(() => mount(
      <BooksItem
        authors={['Matt Haig']}
        images={{
          thumbnail:
            'http://books.google.com/books/content?id=4mZ1XoSpR9UC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
        }}
        title="The Humans"
      />,
  )).toThrow();
});

it('BooksItem throws without series', () => {
  expect(() => mount(
      <BooksItem
        authors={['Matt Haig']}
        images={{
          thumbnail:
            'http://books.google.com/books/content?id=4mZ1XoSpR9UC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
        }}
        isbn={'9780857868770'}
        title="The Humans"
      />,
  )).toThrow();
});

it('BooksItem throws without numberInSeries', () => {
  expect(() => mount(
      <BooksItem
        authors={['Matt Haig']}
        images={{
          thumbnail:
            'http://books.google.com/books/content?id=4mZ1XoSpR9UC&printsec=frontcover&img=1&zoom=1&source=gbs_api',
        }}
        isbn={'9780857868770'}
        series={'The Humans'}
        title="The Humans"
      />,
  )).toThrow();
});
