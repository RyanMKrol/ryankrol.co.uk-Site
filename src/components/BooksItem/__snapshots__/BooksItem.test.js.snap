// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BooksItem without issue 1`] = `
<div
  className="container"
>
  <img
    alt="The Humans - Matt Haig"
    className="thumbnail"
    src="http://books.google.com/books/content?id=4mZ1XoSpR9UC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
  />
</div>
`;
